lib/systems/src/TorqueControllerMux.cpp

```diff
@@ -85,5 +87,94 @@ void TorqueControllerMux::tick(
         }
 
         drivetrainCommand_ = controllerOutputs_[static_cast<int>(muxMode_)].command;
+
+        applyPowerLimit(&drivetrainCommand_);
+        applyTorqueLimit(&drivetrainCommand_);
+        applyPosSpeedLimit(&drivetrainCommand_);
+
+    }
+}
+
+/*
+    Apply power limit such that the mechanical power of all wheels never
+    exceeds the preset mechanical power limit. Scales all wheels down to
+    preserve functionality of torque controllers
+*/
+void TorqueControllerMux::applyPowerLimit(DrivetrainCommand_s* command)
+{
+    float net_torque_mag = 0;
+    float net_wheelspeed = 0;
+
+    for (int i = 0; i < NUM_MOTORS; i++) {
+        // get the total magnitude of torque from all 4 wheels
+        #ifdef ARDUINO_TEENSY41 // screw arduino.h macros
+        net_torque_mag += abs(command->torqueSetpoints[i]);
+        net_wheelspeed += abs(command->speeds_rpm[i]);
+        #else
+        net_torque_mag += std::abs(command->torqueSetpoints[i]);
+        net_wheelspeed += std::abs(command->speeds_rpm[i]);
```

Comment:
Can't use wheel speed target. Need to use actual wheel speed as measured by the inverters.